### PR TITLE
fix: parse WITH command as SELECT

### LINKED ISSUE #5857: [Spanner] `WITH` clause is not recognized as query command type
The Cloud Spanner SQL dialect allows queries to start with either `SELECT` or `WITH`. The latter is currently not recognized as a keyword indicating the start of a `SELECT` command in the client library.

See https://cloud.google.com/spanner/docs/query-syntax for full query syntax reference.

Example query that is a valid Cloud Spanner query, but that is not recognized:

```sql
WITH AlbumsView AS (SELECT * FROM Albums)
select Title
from AlbumsView
```

> I suspect this is what needs changing: https://github.com/googleapis/google-cloud-dotnet/blob/master/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs

### PR DESCRIPTION
**Pull request by olavloite**
Parses `WITH` statements as queries and allows other whitespace characters than only ' '.

Fixes #5857

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs:223]: The documentation suggests that we could use `(char[]) null` instead of `new char[0]`, which would avoid creating an empty array on every call.

**Review by jskeet — APPROVED** [review]
One suggested change, but otherwise looks good.

**Review by amanda-tarafa — COMMENTED** [review]
LGTM. But can you add an integration test that uses WITH? Just to make extra certain.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs
Comment (olavloite) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommandTextBuilder.cs]: Good point, done.

**Comment by olavloite** [comment]
> LGTM. But can you add an integration test that uses WITH? Just to make extra certain.

I've added an integration test case to the existing `ReadTests` class.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/ReadTests.cs
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/ReadTests.cs:602]: Not sure why GitHub isn't showing me a suggestion option, but assuming Spanner doesn't mind line breaks, this can be simplified to:

                cmd.CommandText = $@"
                     WITH LastValue AS (
                        SELECT Key, StringValue
                        FROM {_fixture.TableName}
                        ORDER BY StringValue
                        LIMIT {int.MaxValue} OFFSET {_fixture.RowCount - 1})
                      SELECT * FROM LastValue";

**Review by jskeet — APPROVED** [review]
One suggestion which could be fixed later if it's inconvenient now, but other than that, I'm happy.

**Review by jskeet — APPROVED** [review]
Great, thanks. All looks good to me - let's just wait for @amanda-tarafa to give a final LGTM, then I'll merge.

**Review by amanda-tarafa — APPROVED** [review]
Thanks for the test! LGTM

**Pull request merged**
